cql2pgjson/src/main/java/org/folio/cql2pgjson/CQL2PgJSON.java

```diff
@@ -919,25 +925,66 @@ private String queryByLike(String index, DbIndex dbIndex, IndexTextAndJsonValues
    * @return
    */
   private String queryBySql(DbIndex dbIndex, IndexTextAndJsonValues vals, CQLTermNode node, String comparator, CqlModifiers modifiers) {
-    String index = vals.getIndexText();
-
+    String indexMod = vals.getIndexText();
     if (comparator.equals("==")) {
       comparator = "=";
     }
+    Index schemaIndex = dbIndex.getIndex();
+    String sql;
     String term = "'" + Cql2SqlUtil.cql2like(node.getTerm()) + "'";
     if (CqlTermFormat.NUMBER.equals(modifiers.getCqlTermFormat())) {
-      index = "(" + index + ")::numeric";
-      term = node.getTerm();
+      sql = "(" + indexMod + ")::numeric " + comparator + term;
+    } else if(schemaIndex != null) {
+      sql = createSQLLengthCase(comparator, indexMod, term, false,schemaIndex);
+    } else {
+      sql = indexMod + " " + comparator + term;
     }
-    String sql = index + " " + comparator + term;
 
     if (! dbIndex.hasIndex() && ! dbIndex.hasFullTextIndex() && ! dbIndex.hasLikeIndex()) {
-      String s = String.format("%s, CQL >>> SQL: %s >>> %s", index, node.toCQL(), sql);
+      String s = String.format("%s, CQL >>> SQL: %s >>> %s", indexMod, node.toCQL(), sql);
       logger.log(Level.WARNING, "Doing SQL query without index for {0}", s);
     }
 
-    logger.log(Level.FINE, "index {0} generated SQL {1}", new Object[] {index, sql});
+    logger.log(Level.FINE, "index {0} generated SQL {1}", new Object[] {indexMod, sql});
+    return sql;
+  }
+
+  private String createSQLLengthCase(String comparator, String index, String term, boolean not, Index schemaIndex) {
```

Comment:
The `not` parameter is always false and therefore should be removed.

Reply:
there is one case where the operator is determined by the comparator that is sent in.  On line 963 it checks comparator.equals("<>")